### PR TITLE
Use `withMutations` in reducers to prevent some unnecessary garbage collection, fixes #135

### LINKED ISSUE #135: Consider using `withMutations` in reducers
...so that we don't create a million temporary clones of our state while modifying it.

### PR DESCRIPTION
**Pull request by victorporof**
https://github.com/mozilla/tofino/issues/135

### CODE REVIEW

**Comment by victorporof** [comment]
This is branched off of https://github.com/mozilla/tofino/pull/163 so only 6ebdf44 is relevant here.
@linclark care to review?

**Comment by victorporof** [comment]
Rebased this on top of `master` so that there's no extraneous commits.

File: ui/browser/reducers/main-reducers.js [hunk 1]
```diff
@@ -147,12 +148,18 @@ function setPageOrder(state, pageIndex, updatedIndex) {
     }
   }
 
-  // First remove the page we want to move, then add back to the array,
-  // maybe we want to do this in a faster way?
-  let pages = state.pages.splice(pageIndex, 1);
-  pages = pages.splice(updatedIndex, 0, page);
-  return state.set('pages', pages)
-              .set('currentPageIndex', currentPageIndex);
+  return state.withMutations(mutState => {
```
Comment (jsantell) [inline ui/browser/reducers/main-reducers.js]: s/mutState/mut for consistency

**Comment by jsantell** [comment]
I had some misunderstanding about `withMutations`, but after reading more about it, this makes sense.

**Comment by victorporof** [comment]
Addressed comments.

**Comment by linclark** [comment]
Looks good to me. I don't have privileges to merge, so r+

**Comment by victorporof** [comment]
I'll have to rebase on some new changes made by @bgrins/@jsantell to those reducers. In the future, please make sure you use `withMutations` if there's more than one state mutation in a reducer.

**Pull request closed without merging**